libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -615,25 +615,32 @@ string YulUtilFunctions::divide32CeilFunction()
 string YulUtilFunctions::overflowCheckedIntAddFunction(IntegerType const& _type)
 {
 	string functionName = "checked_add_" + _type.identifier();
-	// TODO: Consider to add a special case for unsigned 256-bit integers
-	//       and use the following instead:
-	//       sum := add(x, y) if lt(sum, x) { <panic>() }
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return
 			Whiskers(R"(
 			function <functionName>(x, y) -> sum {
 				x := <cleanupFunction>(x)
 				y := <cleanupFunction>(y)
+				sum := add(x, y)
 				<?signed>
-					// overflow, if x >= 0 and y > (maxValue - x)
-					if and(iszero(slt(x, 0)), sgt(y, sub(<maxValue>, x))) { <panic>() }
-					// underflow, if x < 0 and y < (minValue - x)
-					if and(slt(x, 0), slt(y, sub(<minValue>, x))) { <panic>() }
+					<?256bit>
+						// overflow, if x >= 0 and sum < y
+						if and(iszero(slt(x, 0)), slt(sum, y)) { <panic>() }
```

Comment:
It might make sense to pull the two ifs into a single one. I think the optimizer cannot do it yet and it could result in more efficient code.

Reply:
This ?
```
if or(
	and(iszero(slt(x, 0)), slt(sum, y)),
	and(slt(x, 0), iszero(slt(sum, y)))
) { <panic>() }
```

Reply:
done